docs/admin/kubeadm.md

```diff
@@ -323,16 +323,60 @@ kind: NodeConfiguration
 caCertPath: <path|string>
 discoveryFile: <path|string>
 discoveryToken: <string>
-
-# Currently only the first server is used as a target for the cluster
-# bootstrap flow.
 discoveryTokenAPIServers:
-  - <address|string>
-  - <address|string>
-
+- <address|string>
+- <address|string>
 tlsBootstrapToken: <string>
 ```
 
+## Securing your installation even more
+
+Most security features are automatically turned on for kubeadm clusters, but there is a balance between usability
+and security. kubeadm clusters start with RBAC and the Node Authorizer turned on, and all communication between
+components secured for instance.
+
+### Turning off auto-approval of Node Client Certificates
+
+However, in order to make the `kubeadm join` flow smooth, there is an CSR auto-approver
+enabled that basically approves any client certificate request for a Node when a Bootstrap Token was used as
+authentication. If you don't want the cluster to automatically approve Node client certs, you can turn it off
+by executing this command:
+
+```console
+$ kubectl delete clusterrole kubeadm:node-autoapprove-bootstrap
+```
+
+After that, `kubeadm join` will block until the admin has manually approved the CSR in flight:
+
+```console
+$ kubectl get csr
+NAME                                                   AGE       REQUESTOR                 CONDITION
+node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ   18s       system:bootstrap:878f07   Pending
+
+$ kubectl certificate approve node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ
+certificatesigningrequest "node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ" approved
+
+$ kubectl get csr
+NAME                                                   AGE       REQUESTOR                 CONDITION
+node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ   1m        system:bootstrap:878f07   Approved,Issued
+```
+
+Only after `kubectl certificate approve` has been run, `kubeadm join` can proceed.
+
+### Turning off public access to the cluster-info ConfigMap
+
+In order to achieve the joining flow using the token as the only piece of validation information, a
+public ConfigMap with some data needed for validation of the master's identity is exposed publicly by
+default. There is no private data in there, but if you anyway want to turn off being able to use the
+`kubeadm join --discovery-token` flow, you can take these steps below:
+
+```
+$ kubectl -n kube-public delete rolebinding kubeadm:bootstrap-signer-clusterinfo
```

Comment:
Perhaps document how to get the `cluster-info` resource in a way that you can just ship that file around manually?  I don't have a cluster up and running but I'm imagining a `kubectl get` with a jsonpath could pull it out.  Then just save that to a file and specify that you pass it to `kubectl join` with the `--discovery-file` param.